File: dashboard_plugin/backend/routes/meta.js
const express = require('express');
const router = express.Router();
const {
    pool
} = require('../db/pool');

// GET issue types
router.get('/issue-types', async (req, res) => {
    try {
        const query = `
      SELECT it.name AS type, isc.name AS subtype
      FROM issue_types it
      LEFT JOIN issue_subtypes isc ON it.issue_type_id = isc.issue_type_id
      ORDER BY it.name, isc.name
    `;
        const result = await pool().query(query);

        const grouped = {};
        result.rows.forEach(row => {
            if (!grouped[row.type]) {
                grouped[row.type] = [];
            }
            if (row.subtype) {
                grouped[row.type].push(row.subtype);
            }
        });
        
        res.json(grouped);
    } catch (err) {
        console.error('Failed to fetch issue categories:', err);
        res.status(500).send('Server error');
    }
});

// GET agencies
router.get('/agencies', async (req, res) => {
    try {
        const result = await pool().query(`SELECT agency_name FROM agencies ORDER BY agency_name`);
        res.json(result.rows.map(r => r.agency_name));
    } catch (err) {
        console.error('Failed to fetch agencies:', err);
        res.status(500).send('Server error');
    }
});

// GET town councils
router.get('/town-councils', async (req, res) => {
    try {
        const result = await pool().query(`SELECT council_name FROM town_councils ORDER BY council_name`);
        res.json(result.rows.map(r => r.council_name));
    } catch (err) {
        console.error('Failed to fetch town councils:', err);
        res.status(500).send('Server error');
    }
});

module.exports = router;

File: dashboard/backend/crud/meta.py
from dashboard.backend.db.database import pool

async def get_issue_types():
    query = """
        SELECT it.name AS type, isc.name AS subtype
        FROM issue_types it
        LEFT JOIN issue_subtypes isc ON it.issue_type_id = isc.issue_type_id
        ORDER BY it.name, isc.name
    """
    async with pool.connection() as conn:
        result = await conn.execute(query)
        rows = await result.fetchall()
    grouped = {}
    for row in rows:
        grouped.setdefault(row["type"], []).append(row["subtype"]) if row["subtype"] else grouped.setdefault(row["type"], [])
    return grouped

async def get_agencies():
    query = "SELECT agency_name FROM agencies ORDER BY agency_name"
    async with pool.connection() as conn:
        result = await conn.execute(query)
        rows = await result.fetchall()
    return [r["agency_name"] for r in rows]

async def get_town_councils():
    query = "SELECT council_name FROM town_councils ORDER BY council_name"
    async with pool.connection() as conn:
        result = await conn.execute(query)
        rows = await result.fetchall()
    return [r["council_name"] for r in rows]
